stl/inc/xutility

```diff
@@ -848,9 +848,25 @@ _INLINE_VAR constexpr bool _Is_input_iter_v = is_convertible_v<_Iter_cat_t<_Iter
 template <class _Iter>
 _INLINE_VAR constexpr bool _Is_fwd_iter_v = is_convertible_v<_Iter_cat_t<_Iter>, forward_iterator_tag>;
 
+template <class _Iter>
+_INLINE_VAR constexpr bool _Is_ranges_fwd_iter_v =
+#if defined(__cpp_lib_concepts)
+    forward_iterator<_Iter>;
+#else
+    is_convertible_v<_Iter_cat_t<_Iter>, forward_iterator_tag>;
+#endif
+
 template <class _Iter>
 _INLINE_VAR constexpr bool _Is_bidi_iter_v = is_convertible_v<_Iter_cat_t<_Iter>, bidirectional_iterator_tag>;
 
+template <class _Iter>
+_INLINE_VAR constexpr bool _Is_ranges_bidi_iter_v =
+#if defined(__cpp_lib_concepts)
+    bidirectional_iterator<_Iter>;
+#else
+    is_convertible_v<_Iter_cat_t<_Iter>, bidirectional_iterator_tag>;
+#endif
+
 template <class _Iter>
 _INLINE_VAR constexpr bool _Is_random_iter_v = is_convertible_v<_Iter_cat_t<_Iter>, random_access_iterator_tag>;

```

Comment:
Are random access iterators already handled?

Reply:
There were no cases where `_Is_ranges_random_iter_v` would have been useful.

Reply:
Some algorithms use `if constexpr` to detect random access iterators and have lower time complexity whenever possible. 
Perhaps we can also do this for `random_access_iterator`, but this may be out of the scope of P2408R5.

Reply:
right, exactly; I don't wish to make any changes not required by the paper.